### ISSUE TITLE
Issue with the experimental results of the paper

### ISSUE THREAD

**Issue by dgjung0220** [xuan611/sRGB-Real-Noise-Synthesizing#6]

**Comment by dgjung0220** [comment]
The experiments in Table 1 cannot be reproduced. I used the provided pretrained weights for evaluation, but the results turned out as follows.
![image](https://github.com/xuan611/sRGB-Real-Noise-Synthesizing/assets/16290330/ebfff26d-1448-4d5c-8141-07961e27e788)

PSNR is low, and KLD is high. Furthermore, as mentioned in the previous issue, even without reimplementing, the averages listed in the paper do not match the numerical values. I used the PSNR and KLD functions provided in your code, and I also in order to maintain consistency between the training and validation sets, validation set contains the same set of ISO levels that you mentioned in the paper.

Furthermore, the denoiser training settings mentioned in the paper (lr=1e-3, epoch 300) are also incorrect. When using your dataloader as is for training, the results for the denoiser performance in terms of PSNR using real noise on the SIDD benchmark reach approximately 38. Table 2 also does not match.

Finally, if all the numbers reported in the experiments are correct, I have one last question. In Table 1, the PSNR for real noise on the validation set is 38.27, and NeCA is 37.58, with a performance gap of 0.69. However, in Table 2, on the SIDD Benchmark set, Real noise is 37.12, and NeCA-W is 36.82, and the performance gap significantly narrows. In my experience, the performance on the SIDD Benchmark set is usually similar to the validation set. However, here, there is a difference of more than 1 dB in PSNR performance between the validation set and the benchmark set. I find this puzzling and question the reliability of the benchmark set.

If I have misunderstood, please provide the correct evaluation code or examination. If you wish, I can provide the code I used for re-implementation using your pretrained weights.

**Comment by zfu006** [comment]
I see, so in a conclusion, you got two questions: (1) KLD and PSNR of released pretrained models msimatch the Table 1 provided in the paper; (2) performance gap mismatches between the validation set and benchmark.

Firstly, I think it is ok if you use your re-evaluate results if you are writing your paper, but I suggest you must retest the table 1 based on the splitted training and testing set for each camera, say you should retest C2N, NLF-ISP. The reason why mismatching occurs is because when I release the code, I modify the way to split training and testing set for evaluating the results in table 1, I add more images into testing set and in this case the images in the training set become less, because I think it is more reasonable. Then based on the new splitted datasets, I trained the model and released the weights. And this is the reason why the PSNR is low and KLD is high since you got more testing set and less training set. You can check the data_preparetion folder to see how I split the training and testing sets. You also can split the sets by using utils.util_image.train_test_split. 

Secondly, the validation set in table 1 is splitted from the training set, and is not the validation set provided by SIDD. If you train a denoiser, the psnr should be same on the validation set and benchmark i in SIDD. However, the validation set used in Table 1 is a part of training set, they not the same.

**Comment by dgjung0220** [comment]
Thanks for your reply. That helps me a lot.
